### PR TITLE
RFC Backend changes - 2

### PR DESCRIPTION
**Pull request by kris-daxiom**
*Issue:*

https://github.com/bcgov/entity/issues/22388
https://github.com/bcgov/entity/issues/22389
https://github.com/bcgov/entity/issues/22350
https://github.com/bcgov/entity/issues/22351

*Description of changes:*

- Added application Id to the LTSA and auto approval model
- Changed the mandatory registration id to optional. Retained it for a possible usecase of multiple properties per application in the future
- Added application ltsa and auto approval endpoints
- Refactored the existing code
- Added unit tests and postman tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: strr-api/src/strr_api/models/auto_approval_record.py [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
-class AutoApprovalRecord(db.Model):
+class AutoApprovalRecord(BaseModel):
     """AutoApprovalRecord Record"""
 
     __tablename__ = "auto_approval_records"
 
     id = db.Column(db.Integer, primary_key=True, autoincrement=True)
-    registration_id = db.Column(db.Integer, db.ForeignKey("registrations.id"), nullable=False)
+    registration_id = db.Column(db.Integer, db.ForeignKey("registrations.id"), nullable=True)
+    application_id = db.Column(db.Integer, db.ForeignKey("application.id"), nullable=True)
```
Comment (shaangill025) [inline strr-api/src/strr_api/models/auto_approval_record.py:21]: Shouldn't `application_id` be required for this model?

File: strr-api/src/strr_api/models/ltsa.py [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
-    """LTSARecord Record"""
+class LTSARecord(BaseModel):
+    """LTSA Record"""
 
     __tablename__ = "ltsa"
 
     id = db.Column(db.Integer, primary_key=True, autoincrement=True)
-    registration_id = db.Column(db.Integer, db.ForeignKey("registrations.id"), nullable=False)
+    registration_id = db.Column(db.Integer, db.ForeignKey("registrations.id"), nullable=True)
+    application_id = db.Column(db.Integer, db.ForeignKey("application.id"), nullable=True)
```
Comment (shaangill025) [inline strr-api/src/strr_api/models/ltsa.py:21]: Same question as above regarding `application_id` being required?

File: strr-api/src/strr_api/models/auto_approval_record.py [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
     record = db.Column(JSONB, nullable=False)
     creation_date = db.Column(db.DateTime, nullable=False, server_default=text("(NOW())"))
+
+    @classmethod
+    def get_registration_auto_approval_records(cls, registration_id: int) -> AutoApprovalRecord:
```
Comment (shaangill025) [inline strr-api/src/strr_api/models/auto_approval_record.py:26]: Can Optional[AutoApprovalRecord] be considered as return type?

**Review by shaangill025 — APPROVED** [review]
LGTM, just listed a few question

**Review by dimak1 — APPROVED** [review]
LGTM!

**Pull request merged**